qiskit/algorithms/linear_solvers/matrices/tridiagonal.py

```diff
@@ -0,0 +1,210 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2019, 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""Hamiltonian simulation of tridiagonal Toeplitz symmetric matrices."""
+
+from typing import Optional
+
+import numpy as np
+from scipy.sparse import diags
+
+from qiskit.circuit import QuantumCircuit, QuantumRegister, AncillaRegister, Qubit
+
+
+class Tridiagonal(QuantumCircuit):
+    """Class of tridiagonal Toeplitz symmetric matrices"""
+
+    def __init__(self, num_state_qubits: int, main_diag: float, off_diag: float,
```

Comment:
then you'd have to input the main and off diagonals as `[main_diag] * num_qubits` because this module is for matrices with a main diagonal consisting of all entries equal to some value, `a`, the two off diagonals of all entries equal to another value, `b`, and everything else `0`.